### PR TITLE
Add task02

### PR DESCRIPTION
**Pull request by julwein**

### CODE REVIEW

File: homework_01/task_02.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    please write function "check_fib", which accepts a Sequence of integers, and
+    returns if the given sequence is a Fibonacci sequence
+We guarantee, that the given sequence contains >= 0 integers inside.
+"""
+from typing import Sequence
+
+
+def gen_fibonacci(n: int) -> Sequence[int]:
+    """Generates Fibonacci sequence from 0 to a number less then or equal to n.
+    NB! If n == 1, the function will return [0, 1, 1]"""
```
Comment (Cjkjvfnby) [inline homework_01/task_02.py]: I don't see any test for this case.

File: homework_01/task_02.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    if n < 0:
+        return []
+    fib_seq = []
+    f0, f1 = 0, 1
+    while f0 <= n:
+        fib_seq.append(f0)
+        f0, f1 = f1, f0 + f1
+    return fib_seq
```
Comment (Cjkjvfnby) [inline homework_01/task_02.py:22]: We don't use the whole list, only part of it. How could we optimize it?

File: test/homework_01/test_task_02.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+from homework_01.task_02 import check_fibonacci
+
+
+def test_positive_case_from_0():
+    """Testing that several consecutive Fibonacci numbers starting with 0 give True"""
+    assert check_fibonacci([0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377])
```
Comment (Cjkjvfnby) [inline test/homework_01/test_task_02.py]: I'd say 5 elements is enough,  try to make test data as small as possible.  I'd say 5 or 10 items work pretty the same for that task.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
I'd say you have solved `given sequence is a Fibonacci SUB-sequence` tasks.  It is a bit harder than just a sequence check.  

Misunderstanding requirements is a quite common problem in development. It leads to extra time to be spent on task.  After you have implemented the task always do an extra look into requirements.

In that case lets continue with solving this harder task.

File: homework_01/task_02.py [hunk 4]
```diff
@@ -7,30 +7,30 @@
...
-        ref_number = reference_seq.pop()
-        if element != ref_number:
-            return False
-    return True
+    beginning = data[0]
+    end = data[-1]
+    ref_seq = gen_fibonacci(beginning, end)
+    # remove duplicate 1 in ref_seq if there's only one 1 in data and
+    # it's at the beginning or at the end of the data list:
+    if (data.count(1) == 1) and (beginning == 1 or end == 1) and (1 in ref_seq):
```
Comment (Cjkjvfnby) [inline homework_01/task_02.py:34]: Could you give an example when `(beginning == 1 or end == 1)` is true and ` (1 in ref_seq)` is false?

File: test/homework_01/test_task_02.py [hunk 5]
```diff
@@ -21,6 +46,16 @@ def test_positive_0_1():
     assert check_fibonacci([0, 1])
 
 
+def test_positive_0_1_1():
+    """Testing that [0, 1, 1]"""
```
Comment (Cjkjvfnby) [inline test/homework_01/test_task_02.py]: that what?

Please remove docstrings from tests and move their content into the test names.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
A couple of cosmetic comments.

File: homework_01/task_02.py (hunk 4 above)
Comment (julwein) [reply homework_01/task_02.py:34]: Yes. `data = [1, -1]` -> `beginning = 1, end = -1` -> `ref_seq = []` -> `(1 in ref_seq)` is false

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**